elements/cc-cuda/install.d/04-cuda

```diff
@@ -16,19 +16,16 @@ export TMPDIR=$NVIDIA_TMP_FOLDER
 
 # compare chroot kernel with running kernel
 CHROOT_KERNEL=$(basename $(ls -td -- /lib/modules/*/ | head -n 1))
-RUNNING_KERNEL=$(uname -r)
-if [ "$CHROOT_KERNEL" != "$RUNNING_KERNEL" ]; then
-	echo "Unmatched kernel!"
-	exit 1	
-fi
 
-# Install kernel headers and wget
-sudo apt-get -qq install -y build-essential linux-headers-$(uname -r) wget
+# Load Nvidia's apt keys
+nvidia_release="${UBUNTU_VERSION//\./}"
+platform="$(uname -i)"
+wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu"$nvidia_release"/"$platform"/cuda-keyring_1.0-1_all.deb
+sudo dpkg -i cuda-keyring_1.0-1_all.deb
 
-# Get the NVIDIA CUDA Toolkit for Ubuntu 16
-wget -nv https://developer.download.nvidia.com/compute/cuda/11.2.0/local_installers/cuda_11.2.0_460.27.04_linux.run
+sudo apt update
 
-# Install the NVIDIA CUDA Toolkit for Ubuntu 16
-sudo sh cuda_11.2.0_460.27.04_linux.run --samples --toolkit --driver --silent --tmpdir $NVIDIA_TMP_FOLDER
+# Install required packages
+sudo apt-get -qq install -y build-essential linux-headers-"$CHROOT_KERNEL" linux-generic-hwe-"$UBUNTU_VERSION" cuda-drivers
```

Comment:
Installing the HWE kernel was frequently a fix for people's GPUs not being picked up by the driver. It does seem that the HWE kernel would overwrite the versioned linux headers. Would it be better to nix the hwe kernel?

Reply:
After taking a look, it appears that `linux-generic-hwe-$VERSION` is a metapackage that installs both the image and the headers
https://packages.ubuntu.com/jammy/linux-generic-hwe-22.04

So if we install this, instead of e.g. `linux-image-generic-hwe-$VERSION`, then we shouldn't install headers manually.